databox/indexer/src/databox/client.ts

```diff
@@ -220,6 +246,25 @@ export class DataboxClient {
         return res.data.id;
     }
 
+    async getOrCreateWorkspaceIdWithSlug(slug: string): Promise<string> {
+        try {
+            return (await this.client.post(
+                `/workspaces`,
+                {
+                    name: slug,
+                    slug: slug,
+                    public: true,
+                    enabledLocales: [],
+                    localeFallbacks: [],
+                    ownerId: getStrict('databox.ownerId')
+                }
+            )).data.id;
+        }
+        catch (e) {
+            return this.getWorkspaceIdFromSlug(slug);
```

Comment:
I would have inversed requests: first GET, if 404 then create

Reply:
habit of pseudo-atomicity... another process could create **between** my-get(failed) / my-create.